From 6058fd6dd1e6931c76ef477fa9d94f5cbda33666 Mon Sep 17 00:00:00 2001
From: Zhao Chenhui <chenhui.zhao@freescale.com>
Date: Fri, 23 Mar 2012 17:03:39 +0800
Subject: [PATCH 360/741] usb: host: enable USB wakeup

commit 3e33d72c6c7639481d331a55b6125f5ffa4bf65d from
http://source.codeaurora.org/external/qoriq/qoriq-components/linux

To enable USB wakeup feature,
 * Add a interrupt flag IRQF_NO_SUSPEND to avoid the USB interrupt
   disabled when suspending.
 * Set the USB module as a wakeup source.

Signed-off-by: Zhao Chenhui <chenhui.zhao@freescale.com>
Signed-off-by: Li Yang <leoli@freescale.com>
Signed-off-by: yinbo.zhu <yinbo.zhu@nxp.com>
Signed-off-by: Zhantao Tang <zhantao.tang@windriver.com>
---
 drivers/usb/host/ehci-fsl.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/drivers/usb/host/ehci-fsl.c b/drivers/usb/host/ehci-fsl.c
index e3d0c1c25160..27386db0fb39 100644
--- a/drivers/usb/host/ehci-fsl.c
+++ b/drivers/usb/host/ehci-fsl.c
@@ -137,7 +137,7 @@ static int fsl_ehci_drv_probe(struct platform_device *pdev)
 
 	/* Don't need to set host mode here. It will be done by tdi_reset() */
 
-	retval = usb_add_hcd(hcd, irq, IRQF_SHARED);
+	retval = usb_add_hcd(hcd, irq, IRQF_SHARED | IRQF_NO_SUSPEND);
 	if (retval != 0)
 		goto err2;
 	device_wakeup_enable(hcd->self.controller);
-- 
2.17.1

